src/Notifier.cpp

```diff
@@ -61,6 +63,24 @@ bool Notifier::Initialize(const size_t handshakeCount) {
 }
 
 /*
+ * Receives message data without caring about handshakes, added sleeping code
+ * to prevent pounding the cpu
+ */
+std::string Notifier::ReceiveData() {
+   StopWatch waitCheck;
+   while (waitCheck.ElapsedSec() < mMaxTimeoutInSec && mKeepRunning.load()) {
```

Comment:
run until timeout makes slow unit tests

the mKeepRunning is so that the notifier can stop the loop relatively quickly after destruction. without adding the mKeepRunning to the conditional, I HAVE to wait timeout seconds before the notifier closes.